### PR TITLE
feat: convert bundle test to TS

### PR DESCRIPTION
**Pull request by arty-name**
## fixes KILTProtocol/ticket#2040
Converted bundle test to TS to improve code quality

## How to test:
yarn bundle && yarn test:bundle

## Checklist:

- [x] I have verified that the code works
- [x] I have verified that the code is easy to understand
  - [ ] If not, I have left a well-balanced amount of inline comments
- [x] I have [left the code in a better state](https://deviq.com/principles/boy-scout-rule)
- [ ] I have documented the changes (where applicable)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -11,7 +11,7 @@
     "build": "yarn workspaces foreach -p --topological-dev --exclude '{root-workspace}' run build",
     "build:docs": "typedoc --theme default --out docs/api --tsconfig tsconfig.docs.json && touch docs/.nojekyll",
     "bundle": "yarn workspace @kiltprotocol/sdk-js run bundle",
-    "clean": "yarn workspaces foreach -p --exclude '{root-workspace}' run clean",
+    "clean": "rm -rf tests/dist && yarn workspaces foreach -p --exclude '{root-workspace}' run clean",
```
Comment (rflechtner) [inline package.json:14]: we're using `rimraf` elsewhere, I think that is to be more platform-agnostic

**Review by rflechtner — COMMENTED** [review]
Given that playwright is already ts enabled, can we not import the test script in the spec file and use [`page.evaluate`](https://playwright.dev/docs/evaluating) to run the code in the browser context instead of adding the script to the html directly? This would have playwright transpile it on the fly, I assume.

**Comment by arty-name** [comment]
Handing off the compilation to the playwright is smart, and I haven’t thought about it. This approach, however, would make it impossible to just open the page in the browser and run the code manually. This was an important tool for me when debugging things. Should I try for the playwright approach anyway?

**Comment by rflechtner** [comment]
> Handing off the compilation to the playwright is smart, and I haven’t thought about it. This approach, however, would make it impossible to just open the page in the browser and run the code manually. This was an important tool for me when debugging things. Should I try for the playwright approach anyway?

You got a point there, and I would not know about debugging because I haven't worked with that code, so you know better if it's an essential tool. I imagine we could retain debugging options if we split the code execution in two; assign the `runAll` function to the window object on the first and run it on the second. This way you'd be able to pause execution in between and play around with the code in the browser, at least when testing headful?

File: package.json (hunk 1 above)
Comment (rflechtner) [reply package.json:14]: yeah I'd say so. I don't know the exact reasons why we're working with `rimraf`, but whatever it solves, it should solve it here too.

**Comment by arty-name** [comment]
I don’t even have the playwright installed, and I can open the html file and run the code now. I feel it would be practical to keep it this way.

File: package.json (hunk 1 above)
Comment (arty-name) [reply package.json]: Done!

**Review by rflechtner — APPROVED** [review]
LGTM!

File: tests/bundle-test.ts [hunk 2]
```diff
@@ -259,9 +289,8 @@ async function runAll() {
   )
   await BlockchainUtils.signAndSubmitTx(authorizedAttTx, devFaucet, {
     resolveOn: BlockchainUtils.IS_IN_BLOCK,
-    reSign: true,
   })
-  if (credential.verify()) {
+  if (await credential.verify()) {
```
Comment (tjwelde) [inline tests/bundle-test.ts:293]: Good catch

**Pull request merged**